### PR TITLE
Added capability for XFEMSingleVariableConstraint to use Functions for jump_flux.

### PR DESCRIPTION
**Pull request by jbtompkins**
Added test to single_var_constraint_2d folder and test demonstrating moving interface capability with two discrete material regions.

ref #13756

### CODE REVIEW

**Comment by jbtompkins** [comment]
Please hold off on merging this if you review it. I need to make some corrections to comments.

File: modules/xfem/include/constraints/XFEMSingleVariableConstraint.h [hunk 1]
```diff
@@ -37,19 +39,19 @@ class XFEMSingleVariableConstraint : public ElemElemConstraint
   /// Vector normal to the internal interface
   Point _interface_normal;
 
-  /// Stabilization parameter in Nitsche's formulation
+  /// Stabilization parameter in Nitsche's formulation and penalty factor in the
+  /// Penalty Method
   Real _alpha;
 
-  /// Vector normal to the internal interface
+  /// Change in variable value ot the interface
```
Comment (jiangwen84) [inline modules/xfem/include/constraints/XFEMSingleVariableConstraint.h]: ot -> at

File: modules/xfem/src/constraints/XFEMSingleVariableConstraint.C [hunk 2]
```diff
@@ -25,24 +26,32 @@ InputParameters
...
 {
   InputParameters params = validParams<ElemElemConstraint>();
-  params.addParam<Real>("alpha", 100, "Stablization parameter in Nitsche's formulation.");
+  params.addParam<Real>("alpha",
+                        100,
+                        "Stabilization parameter in Nitsche's formulation and penalty factor "
+                        "in the Penalty Method. In Nitsche's formulation this should be as "
+                        "small as possible while the method is still stable; while in the "
+                        "Penalty Method you want this to be quite large (e.g. 1e6).");
   params.addParam<Real>("jump", 0, "Jump at the interface.");
```
Comment (jiangwen84) [inline modules/xfem/src/constraints/XFEMSingleVariableConstraint.C]: Can you also change 'jump' from Real to Function?

File: modules/xfem/test/tests/moving_interface/verification/tests [hunk 3]
```diff
@@ -21,6 +21,17 @@
     design = 'LevelSetCutUserObject.md'
     issues = '#12833'
   [../]
+  [./1D_xy_discrete2mat]
+    type = Exodiff
+    input = 1D_xy_discrete2mat.i
+    exodiff = '1D_xy_discrete2mat_out.e 1D_xy_discrete2mat_out.e-s002 1D_xy_discrete2mat_out.e-s003 1D_xy_discrete2mat_out.e-s004 1D_xy_discrete2mat_out.e-s005 1D_xy_discrete2mat_out.e-s006 1D_xy_discrete2mat_out.e-s007'
```
Comment (jiangwen84) [inline modules/xfem/test/tests/moving_interface/verification/tests]: do you need to check exodiff for all the files?

**Review by jiangwen84 — CHANGES_REQUESTED** [review]
Looks good. I have only some minor comments.

File: modules/xfem/src/constraints/XFEMSingleVariableConstraint.C (hunk 2 above)
Comment (jbtompkins) [reply modules/xfem/src/constraints/XFEMSingleVariableConstraint.C]: Yes, I'll do that.

File: modules/xfem/test/tests/moving_interface/verification/tests (hunk 3 above)
Comment (jbtompkins) [reply modules/xfem/test/tests/moving_interface/verification/tests]: As far as I know, in order to make sure there is no change at each timestep, we need to compare all of the files. I believe all the other tests compare all generated exodus files, but if you have another suggestion, I'll change it to whatever you think is best.

Comment (jiangwen84) [reply modules/xfem/test/tests/moving_interface/verification/tests]: Is this test necessary to run 7 time steps?

Comment (jbtompkins) [reply modules/xfem/test/tests/moving_interface/verification/tests]: No, I can reduce the number of steps. How many do you recommend?

**Comment by jbtompkins** [comment]
This is getting a bit busy (as far as number of commits this PR contains). Going to make sure the precheck passes, then close this PR and open a new one with all my commits squashed into one.

**Pull request closed without merging**